File: src/com/l2kt/gameserver/network/clientpackets/AddTradeItem.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.model.World;
import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.model.tradelist.TradeItem;
import com.l2kt.gameserver.model.tradelist.TradeList;
import com.l2kt.gameserver.network.SystemMessageId;
import com.l2kt.gameserver.network.serverpackets.TradeItemUpdate;
import com.l2kt.gameserver.network.serverpackets.TradeOtherAdd;
import com.l2kt.gameserver.network.serverpackets.TradeOwnAdd;

public final class AddTradeItem extends L2GameClientPacket
{
	@SuppressWarnings("unused")
	private int _tradeId;
	private int _objectId;
	private int _count;
	
	public AddTradeItem()
	{
	}
	
	@Override
	protected void readImpl()
	{
		_tradeId = readD();
		_objectId = readD();
		_count = readD();
	}
	
	@Override
	protected void runImpl()
	{
		final Player player = getClient().getActiveChar();
		if (player == null)
			return;
		
		final TradeList trade = player.getActiveTradeList();
		if (trade == null)
			return;
		
		final Player partner = trade.getPartner();
		if (partner == null || World.getInstance().getPlayer(partner.getObjectId()) == null || partner.getActiveTradeList() == null)
		{
			player.sendPacket(SystemMessageId.TARGET_IS_NOT_FOUND_IN_THE_GAME);
			player.cancelActiveTrade();
			return;
		}
		
		if (trade.isConfirmed() || partner.getActiveTradeList().isConfirmed())
		{
			player.sendPacket(SystemMessageId.CANNOT_ADJUST_ITEMS_AFTER_TRADE_CONFIRMED);
			return;
		}
		
		if (!player.getAccessLevel().allowTransaction())
		{
			player.sendPacket(SystemMessageId.YOU_ARE_NOT_AUTHORIZED_TO_DO_THAT);
			player.cancelActiveTrade();
			return;
		}
		
		if (player.validateItemManipulation(_objectId) == null)
		{
			player.sendPacket(SystemMessageId.NOTHING_HAPPENED);
			return;
		}
		
		final TradeItem item = trade.addItem(_objectId, _count);
		if (item != null)
		{
			player.sendPacket(new TradeOwnAdd(item));
			player.sendPacket(new TradeItemUpdate(trade, player));
			trade.getPartner().sendPacket(new TradeOtherAdd(item));
		}
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/AddTradeItem.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.model.World
import com.l2kt.gameserver.network.SystemMessageId
import com.l2kt.gameserver.network.serverpackets.TradeItemUpdate
import com.l2kt.gameserver.network.serverpackets.TradeOtherAdd
import com.l2kt.gameserver.network.serverpackets.TradeOwnAdd

class AddTradeItem : L2GameClientPacket() {
    private var _tradeId: Int = 0
    private var _objectId: Int = 0
    private var _count: Int = 0

    override fun readImpl() {
        _tradeId = readD()
        _objectId = readD()
        _count = readD()
    }

    override fun runImpl() {
        val player = client.activeChar ?: return

        val trade = player.activeTradeList ?: return

        val partner = trade.partner
        if (partner == null || World.getPlayer(partner.objectId) == null || partner.activeTradeList == null) {
            player.sendPacket(SystemMessageId.TARGET_IS_NOT_FOUND_IN_THE_GAME)
            player.cancelActiveTrade()
            return
        }

        if (trade.isConfirmed || partner.activeTradeList!!.isConfirmed) {
            player.sendPacket(SystemMessageId.CANNOT_ADJUST_ITEMS_AFTER_TRADE_CONFIRMED)
            return
        }

        if (!player.accessLevel.allowTransaction) {
            player.sendPacket(SystemMessageId.YOU_ARE_NOT_AUTHORIZED_TO_DO_THAT)
            player.cancelActiveTrade()
            return
        }

        if (player.validateItemManipulation(_objectId) == null) {
            player.sendPacket(SystemMessageId.NOTHING_HAPPENED)
            return
        }

        val item = trade.addItem(_objectId, _count)
        if (item != null) {
            player.sendPacket(TradeOwnAdd(item))
            player.sendPacket(TradeItemUpdate(trade, player))
            trade.partner?.sendPacket(TradeOtherAdd(item))
        }
    }
}